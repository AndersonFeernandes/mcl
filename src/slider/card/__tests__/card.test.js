import "./../../../test.setup";

import React from "react";
import { shallow } from "enzyme";
import Card from "../card";

describe("Testing Card Component", () => {

    it("should render correctly", (done) => {

        const cardDetails = {
            id: 1,
            title: "We are Humans",
            subtitle: "And we love humans",
            text: "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
            image_url: "https://picsum.photos/300/150/?random",
            href: "https://mindera.com/people-and-culture/we-are-humans/",
            is_liked: true
        };

        const wrapper = shallow(
            <Card details={cardDetails} />
        );

        expect(wrapper).toMatchSnapshot();
        done();
    });

    it("should redirect on click", (done) => {

        const cardDetails = {
            id: 1,
            title: "We are Humans",
            subtitle: "And we love humans",
            text: "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
            image_url: "https://picsum.photos/300/150/?random",
            href: "https://mindera.com/people-and-culture/we-are-humans/",
            is_liked: true
        };


        const wrapper = shallow(
            <Card details={cardDetails} />
        );

        global.dom.reconfigure({
            url: cardDetails.href
        });

        wrapper.first().simulate("click");
        expect(window.location.href).toBe(cardDetails.href);
        done();
    });
});