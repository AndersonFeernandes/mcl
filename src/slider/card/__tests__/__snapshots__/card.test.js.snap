// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    details={
        Object {
            "href": "https://mindera.com/people-and-culture/we-are-humans/",
            "id": 1,
            "image_url": "https://picsum.photos/300/150/?random",
            "is_liked": true,
            "subtitle": "And we love humans",
            "text": "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
            "title": "We are Humans",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.header>
          <styled.img
                    alt="Image: We are Humans"
                    src="https://picsum.photos/300/150/?random"
          />
          <styled.img
                    src="badge.png"
          />
          <styled.h1
                    tabIndex={undefined}
          >
                     
                    We are Humans
                     
          </styled.h1>
          <styled.h2
                    tabIndex={undefined}
          >
                     
                    And we love humans
                     
          </styled.h2>
</styled.header>,
        <styled.p
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
                            }
          }
          tabIndex={undefined}
/>,
        <footer>
          <LikeButton
                    id={1}
                    isLiked={true}
                    tabIndex={undefined}
          />
</footer>,
      ],
      "onClick": [Function],
      "onKeyDown": [Function],
      "tabIndex": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              alt="Image: We are Humans"
              src="https://picsum.photos/300/150/?random"
/>,
            <styled.img
              src="badge.png"
/>,
            <styled.h1
              tabIndex={undefined}
>
               
              We are Humans
               
</styled.h1>,
            <styled.h2
              tabIndex={undefined}
>
               
              And we love humans
               
</styled.h2>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "Image: We are Humans",
              "src": "https://picsum.photos/300/150/?random",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": "badge.png",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                "We are Humans",
                " ",
              ],
              "tabIndex": undefined,
            },
            "ref": null,
            "rendered": Array [
              " ",
              "We are Humans",
              " ",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                "And we love humans",
                " ",
              ],
              "tabIndex": undefined,
            },
            "ref": null,
            "rendered": Array [
              " ",
              "And we love humans",
              " ",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
          },
          "tabIndex": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <LikeButton
            id={1}
            isLiked={true}
            tabIndex={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": 1,
            "isLiked": true,
            "tabIndex": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.header>
            <styled.img
                        alt="Image: We are Humans"
                        src="https://picsum.photos/300/150/?random"
            />
            <styled.img
                        src="badge.png"
            />
            <styled.h1
                        tabIndex={undefined}
            >
                         
                        We are Humans
                         
            </styled.h1>
            <styled.h2
                        tabIndex={undefined}
            >
                         
                        And we love humans
                         
            </styled.h2>
</styled.header>,
          <styled.p
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
                                  }
            }
            tabIndex={undefined}
/>,
          <footer>
            <LikeButton
                        id={1}
                        isLiked={true}
                        tabIndex={undefined}
            />
</footer>,
        ],
        "onClick": [Function],
        "onKeyDown": [Function],
        "tabIndex": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                alt="Image: We are Humans"
                src="https://picsum.photos/300/150/?random"
/>,
              <styled.img
                src="badge.png"
/>,
              <styled.h1
                tabIndex={undefined}
>
                 
                We are Humans
                 
</styled.h1>,
              <styled.h2
                tabIndex={undefined}
>
                 
                And we love humans
                 
</styled.h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Image: We are Humans",
                "src": "https://picsum.photos/300/150/?random",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": "badge.png",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  "We are Humans",
                  " ",
                ],
                "tabIndex": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                "We are Humans",
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  "And we love humans",
                  " ",
                ],
                "tabIndex": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                "And we love humans",
                " ",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "We act like humans, we talk like humans, and we think like humans. And we call out anyone who does the opposite.",
            },
            "tabIndex": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LikeButton
              id={1}
              isLiked={true}
              tabIndex={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": 1,
              "isLiked": true,
              "tabIndex": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
